src/views/studio/MyVideosView/MyVideosView.tsx

```diff
@@ -0,0 +1,189 @@
+          description: 'string',
+          isPublic: true,
+          hasMarketing: false,
+          isExplicit: false,
+        })
+      }
+    }
+    createDrafts()
+    return () => {
+      removeAllDrafts()
+    }
+  }, [])
+
+  const isDraftTab = currentTabName === 'Drafts'
+  const isLoading = loading || (videos?.length === 0 && (totalCount ?? 0) > 0)
+  const placeholderItems = Array.from({ length: isLoading ? videosPerPage : 0 }, () => ({
+    id: undefined,
+    progress: undefined,
+  }))
+  const videosWPlaceholders = [...(videos || []), ...placeholderItems]
+  const handleOnResizeGrid = (sizes: number[]) => setVideosPerRow(sizes.length)
+
+  const FetchMoreVideos = (page: number) =>
+    fetchMore({
+      variables: {
+        // substract 1 coz offset index starts at 0
+        offset: videosPerPage * (page - 1),
+        limit: videosPerPage,
+      },
+    })
+  // console.log({ videos, totalCount, hasAnyVideos, isLoading, isPublic_eq })
+  return (
+    <ViewContainer>
+      <StyledText variant="h2">My Videos</StyledText>
+
+      {hasAnyVideos === false ? (
+        <AddVideoView />
+      ) : (
+        <>
+          <TabsContainer>
+            <Tabs initialIndex={0} tabs={[...TABS]} onSelectTab={setCurrentTab} />
+          </TabsContainer>
+          {isDraftTab && (
+            // Should this really be dismissable?
+            <StyledDismissibleMessage
+              id="video-draft-saved-locally-warning"
+              title={'Video Drafts are saved locally'}
+              description={
+                'This mean you can only access one on the device you used to create it. Clearing your browser history will delete all your drafts.'
+              }
+            />
+          )}
+          <Grid maxColumns={null} onResize={handleOnResizeGrid}>
+            {!isDraftTab &&
+              videosWPlaceholders
+                // this makes for a smoother transition between pages
+                .slice(0, videosPerPage)
+                .map((video, idx) => (
+                  <VideoPreviewPublisher
+                    key={idx + '-' + currentTabName + '-' + currentPage}
+                    id={video.id}
+                    showChannel={false}
+                    isLoading={loading}
+                  />
+                ))}
+            {isDraftTab &&
+              drafts
+                // pagination slice
+                .slice(videosPerPage * (currentPage - 1), (currentPage - 1) * videosPerPage + videosPerPage)
+                .map((draft, idx) => (
+                  <VideoPreviewPublisher
+                    key={idx + '-' + currentTabName + '-' + currentPage}
+                    id={draft.id}
+                    showChannel={false}
+                    isDraft
+                    onEditVideoClick={() => ({})}
+                    onDeleteVideoClick={() => {
+                      removeDraft(draft.id)
+                    }}
+                  />
+                ))}
```

Comment:
Let's try to extract it into some variable like `gridContent`, it makes reading the layout too hard